modules/slate-sugar/src/main/ts/ephox/slate-sugar/api/Api.ts

```diff
@@ -0,0 +1,79 @@
+import { Maybe } from '@ephox/katamari';
+
+type path = number[];
+
+interface Editor {}
+
+interface RtcAdaptor {
+  raw: {
+    getModel: () => Maybe<any>;
+    getModelApi: () => Maybe<ModelApi>;
+  };
+}
+
+interface RtcEditor extends Editor {
+  rtcInstance: RtcAdaptor;
+}
+
+interface SlateElement {
+  type: string;
+  children: Array<SlateNode>;
+  class?: string;
+}
+
+interface SlateText {
+  text: string;
+}
+
+type SlateNode = SlateElement | SlateText;
+
+interface List<T>{
+  hd: T;
+  tl: List<T> | 0;
+}
```

Comment:
We can't have this in the API, it's not a shared type, but we already had some plans to make `SlateLoc` use arrays so maybe that can happen separately

Reply:
I wonder what impact it will have on performance switching to arrays but I guess the depth of documents will never be that high that memory reallocation becomes a major bottle neck it should be faster than traversing from the root for every iteration like slate does for its positions. I would really want to hide these implementation details maybe we can prefix them as _ in the reason side then not add any types for it. But opaque types is hard to have in a structural type system so not sure how well this will work. The only thing that is nominal in ts that I know of is enums but tagging everything will be annoying.

Reply:
removed List in a favor of Array (not currently used in Demo)